trunk/src/hpinterp/update_gdof.F90

```diff
@@ -437,3 +440,534 @@ subroutine update_gdof()
    endif
 !
 end subroutine update_gdof
+
+
+
+
+
+
+!-----------------------------------------------------------------------
+!
+!    routine name       - update_gdof_par
+!
+!-----------------------------------------------------------------------
+!> @brief OpenMP parallel version of update_gdof
+!> @date Mar 2023
+!-----------------------------------------------------------------------
+!
+subroutine update_gdof_par
+!
+   use constrained_nodes
+   use data_structure3D
+   use element_data
+   use environment, only: QUIET_MODE
+   use MPI        , only: MPI_COMM_WORLD,MPI_INTEGER,MPI_REAL8,MPI_SUM, &
+                          MPI_MIN,MPI_IN_PLACE,MPI_STATUS_IGNORE
+   use mpi_param  , only: RANK,ROOT,NUM_PROCS
+   use par_mesh   , only: DISTRIBUTED,HOST_MESH
+   use GMP
+!
+   implicit none
+!
+   integer :: ntype
+   character(len=6) :: mdltype
+!
+!..orientation of element nodes
+   integer, dimension(12) :: nedge_orient
+   integer, dimension(6)  :: nface_orient
+!
+!..element nodes
+   integer, dimension(27)      :: nodesl, norientl
+   integer, dimension(MAXNODM) :: nodm
+!
+!..order of approximation for an element
+   integer, dimension(19) :: norder
+!
+!..reference coordinates for an element
+   real(8), dimension(3,8) :: xsub
+!
+!..geometry dofs for an element
+   real(8), dimension(3,MAXbrickH) :: xnod
+!
+!..temporary for OpenMP parallelism
+   real(8) :: coord(3,MAXquadH)
+!
+!..auxiliary variables for timing
+   real(8) :: MPI_Wtime,start_time,end_time
+!
+!..auxiliary variables
+   integer :: iel,iv,ie,ifc,ind,iflag,i,k,loc
+   integer :: mdle,no,nod,nr_elem_nodes,nrnodm,nr_up_elem
+!
+!..additional variables for distributed case
+   integer :: src,rcv,tag,count,ierr,j_loc,j_glb,j_off,loc_max
+   integer :: ndofH,ndofE,ndofV,ndofQ
+   logical :: nod_flg
+   integer :: nod_cnt(NUM_PROCS)
+   integer, allocatable :: nod_loc(:),nod_tmp(:),nod_glb(:),nod_rnk(:)
+   real(8), dimension(:,:), pointer :: buf
+!
+!..OpenMP parallelization
+   integer, allocatable :: nodes_elem(:,:), nodes_irreg(:,:), nrnodes_irreg(:)
+   integer :: nodesi(27)
+   integer :: j, sz, nrnodi
+!
+!-----------------------------------------------------------------------
+!
+   call MPI_BARRIER(MPI_COMM_WORLD, ierr)
+   start_time = MPI_Wtime()
+!
+!$OMP PARALLEL DO
+!..lower the GMP interface flag for all nodes
+   do nod=1,NRNODS
+      NODES(nod)%geom_interf=0
```

Comment:
`geom_interf` not removed yet in threaded version

Reply:
Oops